src/bp/core/sdk/impl.ts

```diff
@@ -88,6 +88,13 @@ export class IOEvent implements sdk.IO.Event {
     }
   }
 
+  private makeId(): string {
+    // nanosecond temporal + 3 rand digits
+    return `${process.hrtime.bigint()}${Math.random()
+      .toString()
+      .slice(-3)}`
+  }
```

Comment:
I am uncomfortable changing the logic for the event ID. Is it really necessary ?

Reply:
That's the whole point of this PR.

1- it's not completely different it just use a more precise timestamp (nanosecond instead of millisecond)
2- date.now() * 1000000 ==> nanosecond
3- I only append 3 random number after

why are you uncomfortable ?

Reply:
Mostly for event ordering, but I ran the proposed change and it seems to be fine, they will still be ordered by time (since there's one more number)

previous vs proposed:

160399212616380900
1276799879874300169